In [1]:
import os
import conda
import pandas as pd
import numpy as np
import json
# from shapely.geometry import Polygon as Poly

pd.options.display.max_columns = 250

conda_file_dir = conda.__file__
conda_dir = conda_file_dir.split('lib')[0]
proj_lib = os.path.join(os.path.join(conda_dir, 'share'), 'proj')
os.environ["PROJ_LIB"] = proj_lib

import matplotlib.pyplot as plt
import matplotlib.cm
import matplotlib.colors as colors

import geopandas
from geopandas.tools import sjoin
import geoplot as gplt
import geoplot.crs as gcrs

import folium

In [ ]:
# %matplotlib inline

Take Land Cover table from EnviroAtlas, regarding land cover, and reconciliate at HUC8 level since chl-a data for TSI estimation is only available at HUC8 spatial scale. Additionaly, the work of Lee and Bakshi, 2019(Energy-Water-CO2 Nexus of Fossil Fuel Based Power Generation, book carbon management) support the use of HUC8 scale for Techno-Ecological synergy (TES) analysis.

The fields of the databse are described below:

| Field Name |                                           Data Layer Name                                           |
|:----------:|:---------------------------------------------------------------------------------------------------:|
|   N_INDEX  |                                      Percent natural land cover                                     |
|    PAGT    | Percentage of WBD 12-digit Hydrologic Unit Land area that is classified as agriculture (21, 81, 82) |
|   PFOR90   |                                  Percent forest and woody wetlands                                  |
|   PWETL95  |                                 Percent emergent herbaceous wetlands                                |
|    PFOR    |                                            Percent forest                                           |
|    PWETL   |                                           Percent wetlands                                          |
|    PDEV    |                                        Percent developed area                                       |
|    PAGC    |                                           Percent cropland                                          |
|    PAGP    |                                           Percent pasture                                           |


HUC12 is transformed in HUC8 for data spatial consistency deleting the last 4 digits of the code (Federal Standards and Procedures for the National Watershaed Boundary Dataset (WBD), USGS, 2013). 

To restimate the land cover percentajes the values of each item for the same HUC8 watershed are summed, and the percentajes recalculated again over the new total value for HUC8

In [12]:
# File loaded again to avoid overwritting of the LandCover_df variable
LandCover_df = pd.read_csv('LandCover/National_metric_tables_in_CSV/CONUS_metrics_May2019_CSV/LandCover.csv') 

# Convert HUC12 to HUC8
LandCover_df['HUC_8'] = LandCover_df['HUC_12'].map(lambda x: str(x)[:-4])
LandCover_df = LandCover_df.drop('HUC_12',1) #0 for rows, 1 for columns

LandCover_df['HUC_8'] = LandCover_df['HUC_8'].astype(int)
LandCover_df['Total'] = 100

LandCover_df

N_INDEX       PFOR      PWETL       PDEV       PAGT       PAGP  \
0      100.000000  81.643204  11.392500   0.000000   0.000000   0.000000   
1      100.000000  74.108200  12.300500   0.000000   0.000000   0.000000   
2      100.000000  78.816101  13.675100   0.000000   0.000000   0.000000   
3      100.000000  72.776901   7.579820   0.000000   0.000000   0.000000   
4      100.000000  74.281403  13.116500   0.000000   0.000000   0.000000   
5       99.839203  80.429001   5.992480   0.160756   0.000000   0.000000   
6      100.000000  82.489899  12.551700   0.000000   0.000000   0.000000   
7      100.000000  79.267097  11.043100   0.000000   0.000000   0.000000   
8       99.994400  80.136299   9.918930   0.000000   0.005647   0.000000   
9       99.774200  83.703796   9.742270   0.225844   0.000000   0.000000   
10     100.000000  79.331200  10.994400   0.000000   0.000000   0.000000   
11      99.851303  82.403503   6.200050   0.148690   0.000000   0.000000   
12     100.000000  69.942001   8.741190   0.000000   0.000000   0.000000   
13     100.000000  77.303398   5.914510   0.000000   0.000000   0.000000   
14     100.000000  82.327400   5.283770   0.000000   0.000000   0.000000   
15     100.000000  82.222900   6.175970   0.000000   0.000000   0.000000   
16      99.981300  73.731300   8.083340   0.018722   0.000000   0.000000   
17     100.000000  75.435204   9.172470   0.000000   0.000000   0.000000   
18     100.000000  78.044899   4.846390   0.000000   0.000000   0.000000   
19     100.000000  82.437500   4.765010   0.000000   0.000000   0.000000   
20     100.000000  81.716202   2.708650   0.000000   0.000000   0.000000   
21     100.000000  83.530403   5.126400   0.000000   0.000000   0.000000   
22     100.000000  77.393097  10.345000   0.000000   0.000000   0.000000   
23     100.000000  82.628899   5.336040   0.000000   0.000000   0.000000   
24     100.000000  87.156403   5.350210   0.000000   0.000000   0.000000   
25      99.996201  84.289803   5.940780   0.000000   0.003806   0.000000   
26     100.000000  83.361900   5.852620   0.000000   0.000000   0.000000   
27      99.994499  79.694901   6.545110   0.000000   0.005535   0.003321   
28     100.000000  82.683403   4.170830   0.000000   0.000000   0.000000   
29      99.973396  81.557602   5.610470   0.000000   0.026571   0.002044   
...           ...        ...        ...        ...        ...        ...   
82885   99.597198   0.000000   0.000000   0.402785   0.000000   0.000000   
82886   24.314501   0.332790   2.445700   9.285660  66.399803  40.708401   
82887   99.969803   0.000000   0.000000   0.000000   0.030206   0.000000   
82888   51.305698   0.084790   0.696361   5.666960  43.027401  21.502899   
82889   11.000500   0.223019   1.795380   8.274570  80.724899  35.530602   
82890   40.889000   0.004915   0.437447   4.949540  54.161499   1.857920   
82891   80.640099   0.006522   0.174457   3.650560  15.709300   0.463046   
82892   92.825104   0.006442   0.118723   7.047010   0.127927   0.000000   
82893   98.236900   0.035152   1.398210   0.994564   0.768499   0.032122   
82894   98.346901   0.018574   0.115424   0.683923   0.969167   0.010614   
82895   93.742599   0.000000   0.063449   0.274372   5.983020   0.354969   
82896   99.694603   0.000000   0.000000   0.006497   0.298864   0.096156   
82897   99.751297   0.000000   0.206619   0.001275   0.247433   0.021682   
82898  100.000000   0.000000   0.093019   0.000000   0.000000   0.000000   
82899   99.719398   0.000000   0.000000   0.280621   0.000000   0.000000   
82900   99.919197   0.000000   0.167314   0.007693   0.073080   0.000000   
82901   65.008598   0.002038   1.229140   2.910460  32.081001   7.885460   
82902   97.132896   0.000000   0.000000   0.327159   2.539940   0.139191   
82903    5.784450   0.006873   0.150052   5.329710  88.885803  41.232300   
82904   41.647900   0.005188   0.285935   3.503280  54.848801  18.875700   
82905   65.604401   0.004695   0.474156   2.745410  31.650

In [13]:
# Grouped and values sumed, but not percetages yet!
LandCover_dfGrouped = LandCover_df.groupby(['HUC_8']).sum()
LandCover_dfGrouped

N_INDEX         PFOR        PWETL         PDEV         PAGT  \
HUC_8                                                                        
1010002    4097.643600  3321.519814   282.046980     2.071450     0.284989   
1010003    2291.215496  1733.613396   297.340113    29.957569   178.826970   
1010004    5987.979889  4391.402512   917.224541    94.488543   517.531725   
1010005    1033.428600   711.407206   244.454398    46.463798   220.107512   
1010006    2194.817802  1476.533783   317.074897     5.078495     0.103808   
1010007    3599.394402  2724.958996   331.773067     0.500214     0.105245   
1010008    1202.609093  1020.640102    80.317040    34.136468    63.254378   
1010009     445.784084   323.178200    76.634000    32.029460   122.186329   
1010010     562.933506   389.770494   138.588140    36.024910   301.041601   
1010011      82.613998    47.460098    27.962299     2.086710    15.299300   
1020001    6160.736687  4706.676411   653.728177    35.523081     3.740213   
1020002    3182.993309  2451.292007   405.574720    12.396440     4.610064   
1020003    4481.313782  3301.240589   869.201635    92.272668   126.413257   
1020004    4431.884109  3481.696198   498.659621    75.427149    92.688715   
1020005    6310.578377  4669.887413  1266.032219   376.571128   412.850645   
1030001    4165.244011  2983.681889   396.700839    29.455575     5.300366   
1030002    2570.966904  1794.169315   202.328960    23.344496     5.688716   
1030003    9467.798759  7654.368603  1254.327705   591.257062  1040.943834   
1040001    3341.541504  2776.653580   206.749762    46.742837    11.715474   
1040002    4882.023380  4193.827991   402.338952   353.213991   364.762655   
1050001    3703.492798  2809.816002   679.753929    70.795757    25.711554   
1050002   10198.365196  6425.135285  2158.193964   465.115620   436.519630   
1050003    3133.351196  2025.690367   651.568851   239.330970   227.318179   
1050004    1866.141113  1140.895431   463.416422    75.812943    58.046144   
1060001    2734.637287  1930.041747   488.801421   571.458448   293.904473   
1060002    4779.526279  4031.867004   451.060711   305.870860   214.602934   
1060003    4728.962828  2943.392588  1109.763688   937.715990   433.321121   
1070001    3009.432899  2835.418007    82.354694   112.004938    78.561902   
1070002    1015.229996   887.241798    73.592999   137.081558    47.688630   
1070003    2318.126808  2013.342796   223.904139   157.622489   124.250430   
...                ...          ...          ...          ...          ...   
18070106   1100.352986   322.148907    13.507125  1197.544273     2.102707   
18070107    375.183800     3.218666     2.803426    21.230790     3.585449   
18070201     22.819670     0.068934     0.696495   175.531197     1.649140   
18070202   1236.624306   192.229874     5.659650   395.182788   168.193133   
18070203   2537.121758   595.805005    24.981554  1575.818078    87.060282   
18070204    106.164231     3.877023     4.409030   388.183601     5.652202   
18070301   1023.154800    25.500259    16.102561   371.298861     5.546324   
18070302   1757.876802   126.386295    22.059405   329.674131   112.448739   
18070303   1198.775455   156.380983    33.720240   663.553286   137.671098   
18070304   2652.305792   184.321247    62.826207  1416.910155    30.784452   
18070305   1462.904675    64.926078    37.786414   127.746081     9.348990   
18080001   2165.800995   332.252342    60.625115    17.457675   216.741187   
18080002   2116.542610   265.387863    25.847944    12.874999    70.582252   
18080003   6895.049286  1605.231828   105.866678   109.431254   195.519766   
18090101   2979.704712   803.857414     7.737157    18.203203     2.092096   
18090102   4175.725410   837.729035    58.354014    57.833967    66.440430   
18090103   2978.295212   401.072203    23.443614    18.022120     3.682500   
18090201   4192.451302   342.439962     1.433104     4.594010     2.954813   
18090202   8046.722496   155.280889  

In [14]:
# Percetages calculation
LandCover_dfHUC8 = LandCover_dfGrouped
LandCover_dfHUC8.update(LandCover_dfHUC8.iloc[:, :-1].div(LandCover_dfHUC8.Total, 0))
LandCover_dfHUC8.update(LandCover_dfHUC8.iloc[:, :-1].mul(100))
LandCover_dfHUC8['Total'] = LandCover_dfHUC8['PFOR'] + LandCover_dfHUC8['PWETL'] + LandCover_dfHUC8['PDEV'] + LandCover_dfHUC8['PAGC'] + LandCover_dfHUC8['PAGP']
LandCover_dfHUC8

N_INDEX       PFOR      PWETL       PDEV       PAGT      PAGP  \
HUC_8                                                                       
1010002   99.942527  81.012678   6.879195   0.050523   0.006951  0.002150   
1010003   91.648620  69.344536  11.893605   1.198303   7.153079  1.563699   
1010004   90.726968  66.536402  13.897342   1.431645   7.841390  1.171523   
1010005   79.494508  54.723631  18.804184   3.574138  16.931347  2.358806   
1010006   99.764446  67.115172  14.412495   0.230841   0.004719  0.003177   
1010007   99.983178  75.693305   9.215919   0.013895   0.002923  0.001090   
1010008   92.508392  78.510777   6.178234   2.625882   4.865721  0.861445   
1010009   74.297347  53.863033  12.772333   5.338243  20.364388  4.674437   
1010010   62.548167  43.307833  15.398682   4.002768  33.449067  4.325463   
1010011   82.613998  47.460098  27.962299   2.086710  15.299300  6.346910   
1020001   99.366721  75.914136  10.544003   0.572953   0.060326  0.036873   
1020002   99.468541  76.602875  12.674210   0.387389   0.144064  0.088657   
1020003   95.347102  70.239161  18.493652   1.963248   2.689644  1.201860   
1020004   96.345307  75.689048  10.840427   1.639721   2.014972  1.528109   
1020005   88.881386  65.773062  17.831440   5.303819   5.814798  4.073629   
1030001   99.172476  71.040045   9.445258   0.701323   0.126199  0.077416   
1030002   98.883342  69.006512   7.781883   0.897865   0.218797  0.112772   
1030003   85.295484  68.958276  11.300250   5.326640   9.377872  7.895679   
1040001   98.280632  81.666282   6.080875   1.374789   0.344573  0.264942   
1040002   87.178989  74.889786   7.184624   6.307393   6.513619  5.787242   
1050001   97.460337  73.942526  17.888261   1.863046   0.676620  0.348597   
1050002   91.877164  57.884102  19.443189   4.190231   3.932609  1.210627   
1050003   87.037533  56.269177  18.099135   6.648083   6.314394  4.950161   
1050004   93.307056  57.044772  23.170821   3.790647   2.902307  1.145841   
1060001   75.962147  53.612271  13.577817  15.873846   8.164013  7.466026   
1060002   90.179741  76.072962   8.510579   5.771148   4.049112  3.091014   
1060003   77.523981  48.252338  18.192847  15.372393   7.103625  6.500448   
1070001   94.044778  88.606813   2.573584   3.500154   2.455059  1.900915   
1070002   84.602500  73.936817   6.132750  11.423463   3.974053  3.454123   
1070003   89.158723  77.436261   8.611698   6.062403   4.778863  4.011015   
...             ...        ...        ...        ...        ...       ...   
18070106  47.841434  14.006474   0.587266  52.067142   0.091422  0.045546   
18070107  93.795950   0.804667   0.700857   5.307697   0.896362  0.890155   
18070201  11.409835   0.034467   0.348248  87.765598   0.824570  0.000000   
18070202  68.701350  10.679437   0.314425  21.954599   9.344063  1.235695   
18070203  60.407661  14.185833   0.594799  37.519478   2.072864  0.915302   
18070204  21.232846   0.775405   0.881806  77.636720   1.130440  0.137331   
18070301  73.082486   1.821447   1.150183  26.521347   0.396166  0.100286   
18070302  79.903491   5.744832   1.002700  14.985188   5.111306  1.330057   
18070303  59.938773   7.819049   1.686012  33.177664   6.883555  0.435105   
18070304  64.690385   4.495640   1.532347  34.558784   0.750840  0.121991   
18070305  91.431542   4.057880   2.361651   7.984130   0.584312  0.056002   
18080001  90.241708  13.843848   2.526046   0.727403   9.030883  7.484397   
18080002  96.206482  12.063085   1.174907   0.585227   3.208284  2.277531   
18080003  95.764573  22.294887   1.470371   1.519879   2.715552  1.854770   
18090101  99.323490  26.795247   0.257905   0.606773   0.069737  0.059253   
18090102  97.109893  19.482071   1.357070   1.344976   1.545126  0.928315   
18090103  99.276507  13.369073   0.781454   0.600737   0.122750  0.104376   
18090201  99.820269   8.153332   0.034122   0.109381   0.070353  0.058073   
18090202  99.342253   1.917048   0.111452   0.517586   0.140162  0.039083   
18090203

In [16]:
# Save the new database
LandCover_dfHUC8.to_csv('DatabasesClean/LandCover_HUC8.csv', index=True)

## 2017 Agricultural Census

The 2017 Agricultural Census is used to determine the distribution of crops on croplands. However, the data provided by the Agricultural Census have a spatial resolution of HUC6 corresponding to each hydrological unit HUC8 to determine the distribution of crops in each portion of crop land corresponding to each hydrological unit HUC8. Area is in acres (2017 census of Agriculture. Watersheds. U.S. Department of Agriculture, 2019).

In [2]:
# Load csv file
AgriCensus_df = pd.read_csv('AgriCensus2017Watershed/wate_all_tablesCLEANED_TRASPOSED.csv')

AgriCensus_df

HUC_6  Year      Corn  Soybeans  Small grains    Cotton  Rice  \
0    H010100  2017    1833.0    1254.0       37038.0       NaN   NaN   
1    H010200  2017    9257.0       NaN        1170.0       NaN   NaN   
2    H010300  2017   14719.0       NaN         293.0       NaN   NaN   
3    H010400  2017    4002.0       NaN           NaN       NaN   NaN   
4    H010500  2017     163.0       NaN           NaN       NaN   NaN   
5    H010600  2017    3639.0       NaN         170.0       NaN   NaN   
6    H010700  2017    4894.0       NaN           NaN       NaN   NaN   
7    H010801  2017   19289.0       NaN           NaN       NaN   NaN   
8    H010802  2017   13153.0     457.0         523.0       NaN   NaN   
9    H010900  2017    4433.0       NaN           6.0       NaN   NaN   
10   H011000  2017   27052.0     896.0         244.0       NaN   NaN   
11   H020200  2017  127581.0   20181.0        9147.0       NaN   NaN   
12   H020301  2017   14230.0   16479.0        4953.0       NaN   NaN   
13   H020302  2017     976.0      31.0        1074.0       NaN   NaN   
14   H020401  2017   87654.0   54297.0       15668.0       NaN   NaN   
15   H020402  2017  215880.0  206982.0       74035.0       NaN   NaN   
16   H020403  2017   62688.0   66589.0       20510.0       NaN   NaN   
17   H020501  2017  217007.0   46799.0       33596.0       NaN   NaN   
18   H020502  2017   97827.0   54011.0       19613.0       NaN   NaN   
19   H020503  2017  600233.0  270941.0      120930.0       NaN   NaN   
20   H020600  2017  269389.0  288411.0      113409.0       NaN   NaN   
21   H020700  2017  360632.0  233096.0      102548.0       NaN   NaN   
22   H020801  2017  302627.0  374561.0      107548.0       NaN   NaN   
23   H020802  2017   60141.0   75966.0       22507.0       NaN   NaN   
24   H030101  2017   61376.0  159918.0       47873.0   57526.0   NaN   
25   H030102  2017  192144.0  455068.0       97392.0  165007.0   NaN   
26   H030201  2017  127083.0  289631.0       55115.0   94866.0   NaN   
27   H030202  2017  137905.0  332801.0       67854.0   50209.0   NaN   
28   H030203  2017   25306.0   35591.0        1980.0    6593.0   NaN   
29   H030300  2017  141058.0  246031.0       65622.0   28424.0   NaN   
..       ...   ...       ...       ...           ...       ...   ...   
341  H190205  2017       NaN       NaN          57.0       NaN   NaN   
342  H190206  2017       NaN       NaN           NaN       NaN   NaN   
343  H190207  2017       NaN       NaN           NaN       NaN   NaN   
344  H190301  2017       NaN       NaN           NaN       NaN   NaN   
345  H190302  2017       NaN       NaN           NaN       NaN   NaN   
346  H190303  2017       NaN       NaN           NaN       NaN   NaN   
347  H190304  2017       NaN       NaN           NaN       NaN   NaN   
348  H190305  2017       NaN       NaN           NaN       NaN   NaN   
349  H190401  2017       NaN       NaN           NaN       NaN   NaN   
350  H190402  2017       NaN       NaN           NaN       NaN   NaN   
351  H190404  2017       NaN       NaN           NaN       NaN   NaN   
352  H190405  2017       NaN       NaN        4509.0       NaN   NaN   
353  H190406  2017       NaN       NaN           NaN       NaN   NaN   
354  H190407  2017       NaN       NaN           NaN       NaN   NaN   
355  H190408  2017       NaN       NaN           NaN       NaN   NaN   
356  H190501  2017       NaN       NaN           NaN       NaN   NaN   
357  H190502  2017       NaN       NaN           NaN       NaN   NaN   
358  H190801  2017       NaN       NaN           NaN       NaN   NaN   
359  H190803  2017       NaN       NaN        5606.0       NaN   NaN   
360  H190804  2017       NaN       NaN           NaN       NaN   NaN   
361  H190901  2017       NaN       NaN           NaN       NaN   NaN   
362  H190902  2017       NaN       NaN           NaN       NaN   NaN   
363  H190903  2017       NaN       NaN           NaN       NaN   NaN   
364  H200100  2017       NaN       NaN         

In [3]:
# AgriCensus_df.dropna(axis='index', how ='any', subset=['HUC_6'])
# AgriCensus_df.dropna(axis='index', how ='any', subset=['HUC_6']).to_csv('AgriCensus2017Watershed/wate_all_tablesCLEANED_TRASPOSED.csv',index=False)

In [5]:
# Percetages calculation
AgriCensus_df['Total'] = AgriCensus_df[['Corn', 'Soybeans', 'Small grains', 'Cotton', 'Rice', 'Vegetables', 'Orchards', 'Greenhouse', 'Other']].sum(axis=1)
# AgriCensus_df.iloc[:, 1:].astype(float)

AgriCensus_df

HUC_6  Year      Corn  Soybeans  Small grains    Cotton  Rice  \
0    H010100  2017    1833.0    1254.0       37038.0       NaN   NaN   
1    H010200  2017    9257.0       NaN        1170.0       NaN   NaN   
2    H010300  2017   14719.0       NaN         293.0       NaN   NaN   
3    H010400  2017    4002.0       NaN           NaN       NaN   NaN   
4    H010500  2017     163.0       NaN           NaN       NaN   NaN   
5    H010600  2017    3639.0       NaN         170.0       NaN   NaN   
6    H010700  2017    4894.0       NaN           NaN       NaN   NaN   
7    H010801  2017   19289.0       NaN           NaN       NaN   NaN   
8    H010802  2017   13153.0     457.0         523.0       NaN   NaN   
9    H010900  2017    4433.0       NaN           6.0       NaN   NaN   
10   H011000  2017   27052.0     896.0         244.0       NaN   NaN   
11   H020200  2017  127581.0   20181.0        9147.0       NaN   NaN   
12   H020301  2017   14230.0   16479.0        4953.0       NaN   NaN   
13   H020302  2017     976.0      31.0        1074.0       NaN   NaN   
14   H020401  2017   87654.0   54297.0       15668.0       NaN   NaN   
15   H020402  2017  215880.0  206982.0       74035.0       NaN   NaN   
16   H020403  2017   62688.0   66589.0       20510.0       NaN   NaN   
17   H020501  2017  217007.0   46799.0       33596.0       NaN   NaN   
18   H020502  2017   97827.0   54011.0       19613.0       NaN   NaN   
19   H020503  2017  600233.0  270941.0      120930.0       NaN   NaN   
20   H020600  2017  269389.0  288411.0      113409.0       NaN   NaN   
21   H020700  2017  360632.0  233096.0      102548.0       NaN   NaN   
22   H020801  2017  302627.0  374561.0      107548.0       NaN   NaN   
23   H020802  2017   60141.0   75966.0       22507.0       NaN   NaN   
24   H030101  2017   61376.0  159918.0       47873.0   57526.0   NaN   
25   H030102  2017  192144.0  455068.0       97392.0  165007.0   NaN   
26   H030201  2017  127083.0  289631.0       55115.0   94866.0   NaN   
27   H030202  2017  137905.0  332801.0       67854.0   50209.0   NaN   
28   H030203  2017   25306.0   35591.0        1980.0    6593.0   NaN   
29   H030300  2017  141058.0  246031.0       65622.0   28424.0   NaN   
..       ...   ...       ...       ...           ...       ...   ...   
341  H190205  2017       NaN       NaN          57.0       NaN   NaN   
342  H190206  2017       NaN       NaN           NaN       NaN   NaN   
343  H190207  2017       NaN       NaN           NaN       NaN   NaN   
344  H190301  2017       NaN       NaN           NaN       NaN   NaN   
345  H190302  2017       NaN       NaN           NaN       NaN   NaN   
346  H190303  2017       NaN       NaN           NaN       NaN   NaN   
347  H190304  2017       NaN       NaN           NaN       NaN   NaN   
348  H190305  2017       NaN       NaN           NaN       NaN   NaN   
349  H190401  2017       NaN       NaN           NaN       NaN   NaN   
350  H190402  2017       NaN       NaN           NaN       NaN   NaN   
351  H190404  2017       NaN       NaN           NaN       NaN   NaN   
352  H190405  2017       NaN       NaN        4509.0       NaN   NaN   
353  H190406  2017       NaN       NaN           NaN       NaN   NaN   
354  H190407  2017       NaN       NaN           NaN       NaN   NaN   
355  H190408  2017       NaN       NaN           NaN       NaN   NaN   
356  H190501  2017       NaN       NaN           NaN       NaN   NaN   
357  H190502  2017       NaN       NaN           NaN       NaN   NaN   
358  H190801  2017       NaN       NaN           NaN       NaN   NaN   
359  H190803  2017       NaN       NaN        5606.0       NaN   NaN   
360  H190804  2017       NaN       NaN           NaN       NaN   NaN   
361  H190901  2017       NaN       NaN           NaN       NaN   NaN   
362  H190902  2017       NaN       NaN           NaN       NaN   NaN   
363  H190903  2017       NaN       NaN           NaN       NaN   NaN   
364  H200100  2017       NaN       NaN         

In [6]:
AgriCensus_df.update(AgriCensus_df.iloc[:, 2:-1].div(AgriCensus_df.Total, 0))
AgriCensus_df.update(AgriCensus_df.iloc[:, 2:-1].mul(100))
AgriCensus_df['Sum'] = AgriCensus_df.iloc[:, 2:-1].sum(axis=1)
AgriCensus_df

HUC_6  Year       Corn   Soybeans  Small grains     Cotton  Rice  \
0    H010100  2017   1.603983   1.097324     32.410438        NaN   NaN   
1    H010200  2017  19.464244        NaN      2.460102        NaN   NaN   
2    H010300  2017  19.461339        NaN      0.387402        NaN   NaN   
3    H010400  2017  13.170973        NaN           NaN        NaN   NaN   
4    H010500  2017   0.283779        NaN           NaN        NaN   NaN   
5    H010600  2017   7.383735        NaN      0.344940        NaN   NaN   
6    H010700  2017   8.926910        NaN           NaN        NaN   NaN   
7    H010801  2017  15.850281        NaN           NaN        NaN   NaN   
8    H010802  2017  13.206884   0.458872      0.525143        NaN   NaN   
9    H010900  2017   7.981635        NaN      0.010803        NaN   NaN   
10   H011000  2017  23.738362   0.786248      0.214112        NaN   NaN   
11   H020200  2017  22.430720   3.548133      1.608185        NaN   NaN   
12   H020301  2017  16.190509  18.749360      5.635389        NaN   NaN   
13   H020302  2017   4.974009   0.157986      5.473448        NaN   NaN   
14   H020401  2017  25.917877  16.054749      4.632775        NaN   NaN   
15   H020402  2017  30.438087  29.183510     10.438594        NaN   NaN   
16   H020403  2017  33.293324  35.365128     10.892772        NaN   NaN   
17   H020501  2017  22.529071   4.858544      3.487845        NaN   NaN   
18   H020502  2017  31.241657  17.248747      6.263533        NaN   NaN   
19   H020503  2017  40.981494  18.498761      8.256614        NaN   NaN   
20   H020600  2017  35.790689  38.317928     15.067379        NaN   NaN   
21   H020700  2017  26.697404  17.255978      7.591576        NaN   NaN   
22   H020801  2017  30.581111  37.850196     10.867957        NaN   NaN   
23   H020802  2017  14.297193  18.059237      5.350542        NaN   NaN   
24   H030101  2017   9.113804  23.746436      7.108725   8.542112   NaN   
25   H030102  2017  18.635127  44.134868      9.445584  16.003239   NaN   
26   H030201  2017  18.083388  41.213300      7.842638  13.499042   NaN   
27   H030202  2017  18.316364  44.202198      9.012280   6.668694   NaN   
28   H030203  2017  33.538315  47.169136      2.624115   8.737774   NaN   
29   H030300  2017  19.943277  34.784729      9.277870   4.018685   NaN   
..       ...   ...        ...        ...           ...        ...   ...   
341  H190205  2017        NaN        NaN      0.827646        NaN   NaN   
342  H190206  2017        NaN        NaN           NaN        NaN   NaN   
343  H190207  2017        NaN        NaN           NaN        NaN   NaN   
344  H190301  2017        NaN        NaN           NaN        NaN   NaN   
345  H190302  2017        NaN        NaN           NaN        NaN   NaN   
346  H190303  2017        NaN        NaN           NaN        NaN   NaN   
347  H190304  2017        NaN        NaN           NaN        NaN   NaN   
348  H190305  2017        NaN        NaN           NaN        NaN   NaN   
349  H190401  2017        NaN        NaN           NaN        NaN   NaN   
350  H190402  2017        NaN        NaN           NaN        NaN   NaN   
351  H190404  2017        NaN        NaN           NaN        NaN   NaN   
352  H190405  2017        NaN        NaN     37.874843        NaN   NaN   
353  H190406  2017        NaN        NaN           NaN        NaN   NaN   
354  H190407  2017        NaN        NaN           NaN        NaN   NaN   
355  H190408  2017        NaN        NaN           NaN        NaN   NaN   
356  H190501  2017        NaN        NaN           NaN        NaN   NaN   
357  H190502  2017        NaN        NaN           NaN        NaN   NaN   
358  H190801  2017        NaN        NaN           NaN        NaN   NaN   
359  H190803  2017        NaN        NaN     29.879544        NaN   NaN   
360  H190804  2017        NaN        NaN           NaN        NaN   NaN   
361  H190901  2017        NaN        NaN           NaN        NaN   NaN   
362  H190902  2017        NaN   

In [8]:
# Delete H from HUC_6 columns
AgriCensus_df['HUC_6'] = AgriCensus_df['HUC_6'].map(lambda x: str(x)[1:])
AgriCensus_df

HUC_6  Year       Corn   Soybeans  Small grains     Cotton  Rice  \
0    010100  2017   1.603983   1.097324     32.410438        NaN   NaN   
1    010200  2017  19.464244        NaN      2.460102        NaN   NaN   
2    010300  2017  19.461339        NaN      0.387402        NaN   NaN   
3    010400  2017  13.170973        NaN           NaN        NaN   NaN   
4    010500  2017   0.283779        NaN           NaN        NaN   NaN   
5    010600  2017   7.383735        NaN      0.344940        NaN   NaN   
6    010700  2017   8.926910        NaN           NaN        NaN   NaN   
7    010801  2017  15.850281        NaN           NaN        NaN   NaN   
8    010802  2017  13.206884   0.458872      0.525143        NaN   NaN   
9    010900  2017   7.981635        NaN      0.010803        NaN   NaN   
10   011000  2017  23.738362   0.786248      0.214112        NaN   NaN   
11   020200  2017  22.430720   3.548133      1.608185        NaN   NaN   
12   020301  2017  16.190509  18.749360      5.635389        NaN   NaN   
13   020302  2017   4.974009   0.157986      5.473448        NaN   NaN   
14   020401  2017  25.917877  16.054749      4.632775        NaN   NaN   
15   020402  2017  30.438087  29.183510     10.438594        NaN   NaN   
16   020403  2017  33.293324  35.365128     10.892772        NaN   NaN   
17   020501  2017  22.529071   4.858544      3.487845        NaN   NaN   
18   020502  2017  31.241657  17.248747      6.263533        NaN   NaN   
19   020503  2017  40.981494  18.498761      8.256614        NaN   NaN   
20   020600  2017  35.790689  38.317928     15.067379        NaN   NaN   
21   020700  2017  26.697404  17.255978      7.591576        NaN   NaN   
22   020801  2017  30.581111  37.850196     10.867957        NaN   NaN   
23   020802  2017  14.297193  18.059237      5.350542        NaN   NaN   
24   030101  2017   9.113804  23.746436      7.108725   8.542112   NaN   
25   030102  2017  18.635127  44.134868      9.445584  16.003239   NaN   
26   030201  2017  18.083388  41.213300      7.842638  13.499042   NaN   
27   030202  2017  18.316364  44.202198      9.012280   6.668694   NaN   
28   030203  2017  33.538315  47.169136      2.624115   8.737774   NaN   
29   030300  2017  19.943277  34.784729      9.277870   4.018685   NaN   
..      ...   ...        ...        ...           ...        ...   ...   
341  190205  2017        NaN        NaN      0.827646        NaN   NaN   
342  190206  2017        NaN        NaN           NaN        NaN   NaN   
343  190207  2017        NaN        NaN           NaN        NaN   NaN   
344  190301  2017        NaN        NaN           NaN        NaN   NaN   
345  190302  2017        NaN        NaN           NaN        NaN   NaN   
346  190303  2017        NaN        NaN           NaN        NaN   NaN   
347  190304  2017        NaN        NaN           NaN        NaN   NaN   
348  190305  2017        NaN        NaN           NaN        NaN   NaN   
349  190401  2017        NaN        NaN           NaN        NaN   NaN   
350  190402  2017        NaN        NaN           NaN        NaN   NaN   
351  190404  2017        NaN        NaN           NaN        NaN   NaN   
352  190405  2017        NaN        NaN     37.874843        NaN   NaN   
353  190406  2017        NaN        NaN           NaN        NaN   NaN   
354  190407  2017        NaN        NaN           NaN        NaN   NaN   
355  190408  2017        NaN        NaN           NaN        NaN   NaN   
356  190501  2017        NaN        NaN           NaN        NaN   NaN   
357  190502  2017        NaN        NaN           NaN        NaN   NaN   
358  190801  2017        NaN        NaN           NaN        NaN   NaN   
359  190803  2017        NaN        NaN     29.879544        NaN   NaN   
360  190804  2017        NaN        NaN           NaN        NaN   NaN   
361  190901  2017        NaN        NaN           NaN        NaN   NaN   
362  190902  2017        NaN        NaN           NaN        NaN   NaN   
363  19090

In [10]:
# Save the new database
AgriCensus_df.to_csv('DatabasesClean/Agricensus.csv', index=False)

## Vegetation nutrient uptake

To determine the nutrients uptake of each type of crop data from USDA (2009) is considered (Waste Management Field Handbook). 

For croplands, the land is divided in the portion occupied by corn, soybeans, small grains, cotton, rice,	vegetables, orchards, greenhouse and other crops (namely oil crops, sugar crops, and fruits). 

For pasture lands the average nutrient uptake and crop yield including the following plants is considered:

    -Alfalfa
    -Switchgrass
    -Wheatgrass

For forests lands the nutrient uptake and crop yield of Northern hardwoods is considered

For corn, soybeans, cotton, rice and orchards their specific nutrient uptake values are used. For small grains, vegetables, greenhouse crops, pasture crops and forest average values including the most representatve species are used. (USDA, 2009. Waste Management Field Handbook)b

In [33]:
PlantNutrientUptake_df =  pd.read_csv('PlantNutrientUptake/PlantNutrientUptake.csv', index_col='Crop')
PlantNutrientUptake_df

ConversionFactor  YieldAcre  PoundsAcre     N     P     K  \
Crop                                                                        
Barley                      48.0      50.00      2400.0  1.82  0.34  0.43   
BarleyStraw               2000.0       1.00      2000.0  0.75  0.11  1.25   
Buckwheat                   48.0      30.00      1440.0  1.65  0.31  0.45   
BuckwheatStraw            2000.0       0.50      1000.0  0.78  0.05  2.26   
Corn                        56.0     120.00      6720.0  1.61  0.28  0.40   
CornStraw                 2000.0       1.50      3000.0  1.11  0.20  1.34   
Oats                        32.0      80.00      2560.0  1.95  0.34  0.49   
OatsStraw                 2000.0       2.00      4000.0  0.63  0.16  1.66   
Rice                        45.0    5500.00    247500.0  1.39  0.24  0.23   
RiceStraw                 2000.0       2.50      5000.0  0.60  0.09  1.16   
Rye                         56.0      30.00      1680.0  2.08  0.26  0.49   
RyeStraw                  2000.0       1.50      3000.0  0.50  0.12  0.69   
Sorghum                     56.0      60.00      3360.0  1.67  0.36  0.42   
SorghumStraw              2000.0       3.00      6000.0  1.08  0.15  1.31   
Wheat                       60.0      40.00      2400.0  2.08  0.62  0.52   
WheatStraw                2000.0       1.50      3000.0  0.67  0.07  0.97   
Flax                        56.0      15.00       840.0  4.09  0.55  0.84   
FlaxStraw                 2000.0       1.75      3500.0  1.24  0.11  1.75   
Oil palm                     1.0   22000.00     22000.0  1.13  0.26  0.16   
Oil palmStraw             2000.0       5.00     10000.0  1.07  0.49  1.69   
Peanuts                     26.0    2800.00     72800.0  3.60  0.17  0.50   
PeanutsStraw              2000.0       2.20      4400.0  2.33  0.24  1.75   
Rapeseed                    50.0      35.00      1750.0  3.60  0.79  0.76   
RapeseedStraw             2000.0       3.00      6000.0  4.48  0.43  3.37   
Soybeans                    60.0      35.00      2100.0  6.25  0.64  1.90   
SoybeansStraw             2000.0       2.00      4000.0  2.25  0.22  1.04   
Sunflower                   25.0    1100.00     27500.0  3.57  1.71  1.11   
SunflowerStraw            2000.0       4.00      8000.0  1.50  0.18  2.92   
Cotton                       1.0     600.00       600.0  2.67  0.58  0.83   
CottonStraw                  1.0    1000.00      1000.0  1.75  0.22  1.45   
...                          ...        ...         ...   ...   ...   ...   
Sorghum-sudan             2000.0       5.00     10000.0  1.36  0.16  1.45   
Sugarcane                 2000.0       0.16       320.0  0.04  0.37  0.05   
Sugar beets               2000.0       0.20       400.0  0.03  0.14  0.11   
Tobacco                      1.0    2100.00      2100.0  3.75  0.33  4.98   
Bluegrass                 2000.0       2.00      4000.0  2.91  0.43  1.95   
Bentgrass                 2000.0       2.50      5000.0  3.10  0.41  2.21   
Bermudagrass              2000.0       4.00      8000.0  1.88  0.19  1.40   
Bell peppers              2000.0       9.00     18000.0  0.40  0.12  0.49   
Beans                     2000.0       0.50      1000.0  3.13  0.45  0.86   
Cabbage                   2000.0      20.00     40000.0  0.33  0.04  0.27   
Carrots                   2000.0      13.00     26000.0  0.19  0.04  0.25   
Cassava                   2000.0       7.00     14000.0  0.40  0.13  0.63   
Celery                    2000.0      27.00     54000.0  0.17  0.09  0.45   
Cucumbers                 2000.0      10.00     20000.0  0.20  0.07  0.33   
Lettuce                   2000.0      14.00     28000.0  0.23  0.08  0.46   
Onions                    2000.0      18.00     36000.0  0.30  0.06  0.22   
Peas                      2000.0       1.50      3000.0  3.68  0.40  0.90   
Potatoes                  2000.0      14.50     29000.0  0.33  0.06  0.52   
Snap beans                2000.0       3.00      6000.0  0.88  0.26  0.96   
Sweet corn  

For pasture lands the average nutrient uptake and crop yield including the following plants is considered:

    -Alfalfa
    -Switchgrass
    -Wheatgrass

In [34]:
Pasture_df = PlantNutrientUptake_df.loc[['Alfalfa','Switchgrass','Wheatgrass']]
Pasture_df = Pasture_df.mean(axis=0)
Pasture_df.to_csv('DatabasesClean/PasturePlantNutrientUptake.csv', index=False)
Pasture_df

/home/emh/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:3: FutureWarning: The signature of `Series.to_csv` was aligned to that of `DataFrame.to_csv`, and argument 'header' will change its default value from False to True: please pass an explicit value to suppress this warning.
  This is separate from the ipykernel package so we can avoid doing imports until


ConversionFactor    2000.000000
YieldAcre              2.666667
PoundsAcre          5333.333333
N                      1.606667
P                      0.196667
K                      2.150000
Ca                     0.680000
Mg                     0.250000
S                      0.175000
Cu                     0.000800
Mn                     0.005500
Zn                     0.005300
dtype: float64

For forests lands the nutrient uptake and crop yield of Northern hardwoods is considered

In [35]:
Forests_df = PlantNutrientUptake_df.loc[['Northern hardwoods']]
Forests_df = Forests_df.mean(axis=0)
Forests_df.to_csv('DatabasesClean/ForestsPlantNutrientUptake.csv', index=False)
Forests_df

/home/emh/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:3: FutureWarning: The signature of `Series.to_csv` was aligned to that of `DataFrame.to_csv`, and argument 'header' will change its default value from False to True: please pass an explicit value to suppress this warning.
  This is separate from the ipykernel package so we can avoid doing imports until


ConversionFactor      2000.00
YieldAcre               50.00
PoundsAcre          100000.00
N                        0.20
P                        0.02
K                        0.10
Ca                       0.29
Mg                        NaN
S                         NaN
Cu                        NaN
Mn                        NaN
Zn                        NaN
dtype: float64

For small grains the average nutrient uptake and crop yield including the following plants is considered:

    -Wheat
    -WheatStraw
    -Oats
    -OatsStraw
    -Barley
    -BarleyStraw
    -Rye
    -RyeStraw

In [36]:
SmallGrains_df = PlantNutrientUptake_df.loc[['Wheat','WheatStraw','Oats','OatsStraw','Barley','BarleyStraw','Rye','RyeStraw']]
SmallGrains_df = SmallGrains_df.mean(axis=0)
SmallGrains_df.to_csv('DatabasesClean/SmallGrainsPlantNutrientUptake.csv', index=False)
SmallGrains_df

/home/emh/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:3: FutureWarning: The signature of `Series.to_csv` was aligned to that of `DataFrame.to_csv`, and argument 'header' will change its default value from False to True: please pass an explicit value to suppress this warning.
  This is separate from the ipykernel package so we can avoid doing imports until


ConversionFactor    1024.500000
YieldAcre             25.750000
PoundsAcre          2630.000000
N                      1.310000
P                      0.252500
K                      0.812500
Ca                     0.170000
Mg                     0.140000
S                      0.201250
Cu                     0.004613
Mn                     0.006525
Zn                     0.003300
dtype: float64

For vegetables and greehouse the average nutrient uptake and crop yield including the following plants is considered:

    -Bell peppers
    -Beans
    -Cabbage
    -Carrots
    -Cassava
    -Celery
    -Cucumbers
    -Lettuce
    -Onions
    -Peas
    -Potatoes
    -Snap beans
    -Sweet corn
    -Sweet potatoes
    -Table beets
    -Cantaloupe

In [37]:
Vegetables_Greenhouse_df = PlantNutrientUptake_df.loc[['Bell peppers','Beans','Cabbage','Carrots','Cassava','Celery','Cucumbers','Lettuce',
                                                       'Onions','Peas','Potatoes','Snap beans','Sweet corn','Sweet potatoes','Table beets',
                                                       'Cantaloupe']]
Vegetables_Greenhouse_df = Vegetables_Greenhouse_df.mean(axis=0)
Vegetables_Greenhouse_df.to_csv('DatabasesClean/Vegetables_GreenhousePlantNutrientUptake.csv', index=False)
Vegetables_Greenhouse_df

/home/emh/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:5: FutureWarning: The signature of `Series.to_csv` was aligned to that of `DataFrame.to_csv`, and argument 'header' will change its default value from False to True: please pass an explicit value to suppress this warning.
  """


ConversionFactor     1875.062500
YieldAcre            1104.062500
PoundsAcre          21718.750000
N                       0.744375
P                       0.138125
K                       0.505000
Ca                      0.071000
Mg                      0.084286
S                       0.096000
Cu                      0.000275
Mn                      0.001175
Zn                      0.001040
dtype: float64

For other crops, the average nutrient uptake and crop yield considered includes the most representative crop types not considered in other items, namely oil crops, sugar crops, and fruits:

    -Flax
    -FlaxStraw
    -Oil palm
    -Oil palmStraw
    -Peanuts
    -PeanutsStraw
    -Rapeseed
    -RapeseedStraw
    -Soybeans
    -SoybeansStraw
    -Sunflower
    -SunflowerStraw
    -Apples
    -Bananas
    -Coconuts
    -Grapes
    -Oranges
    -Peaches
    -Pineapple
    -Tomatoes
    -Sugarcane

In [38]:
OtherCrops_df = PlantNutrientUptake_df.loc[['Flax','FlaxStraw','Oil palm','Oil palmStraw','Peanuts','PeanutsStraw','Rapeseed','RapeseedStraw',
                                            'Soybeans','SoybeansStraw','Sunflower','SunflowerStraw','Apples','Bananas','Coconuts','Grapes',
                                            'Oranges','Peaches','Pineapple','Tomatoes','Sugarcane']]
OtherCrops_df = OtherCrops_df.mean(axis=0)
OtherCrops_df.to_csv('DatabasesClean/OtherCropsPlantNutrientUptake.csv', index=False)
OtherCrops_df

/home/emh/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:5: FutureWarning: The signature of `Series.to_csv` was aligned to that of `DataFrame.to_csv`, and argument 'header' will change its default value from False to True: please pass an explicit value to suppress this warning.
  """


ConversionFactor     1248.571429
YieldAcre            4284.838571
PoundsAcre          18290.952381
N                       1.990476
P                       0.349524
K                       1.180000
Ca                      0.415000
Mg                      0.219048
S                       0.194667
Cu                      0.001067
Mn                      0.006575
Zn                      0.002614
dtype: float64

## Wetlands nutrients regulation

The phosporus uptake due to wetlands is considered as 0.77 gP·m-2·year-1 based in the data reported by Kadlec, 2016 (Kadlec, R.H., 2016
 Large Constructed Wetlands for Phosphorus Control: A Review. Water, 8, 243)